pallets/marketplace/src/lib.rs

```diff
@@ -559,24 +595,18 @@ pub mod pallet {
             let assignment =
                 <StoredMatches<T>>::get(&who, &job_id).ok_or(Error::<T>::JobNotAssigned)?;
 
-            let now = Self::now()?
-                .checked_add(T::ReportTolerance::get())
-                .ok_or(Error::<T>::CalculationOverflow)?;
-            let (_actual_start, actual_end) = registration
-                .schedule
-                .range(assignment.start_delay)
-                .ok_or(Error::<T>::CalculationOverflow)?;
-            ensure!(actual_end.lt(&now), Error::<T>::JobCannotBeFinalized);
+            ensure!(
+                Self::actual_schedule_ended(&registration.schedule, &assignment)?,
+                Error::<T>::JobCannotBeFinalized
+            );
+
+            let unmet: u64 = assignment.sla.total - assignment.sla.met;
 
             // update reputation since we don't expect further reports for this job
             // (only update for attested devices!)
             if ensure_source_verified::<T>(&who).is_ok() {
-                let extra: <T as Config>::RegistrationExtra = registration.extra.clone().into();
-                let requirements: JobRequirementsFor<T> = extra.into();
-
                 // skip reputation update if reward is 0
-                let reward_amount: <T as Config>::Balance = requirements.reward.into();
-                if reward_amount > 0u8.into() {
+                if assignment.fee_per_execution > 0u8.into() {
```

Comment:
The name `fee_per_execution` is confusing, it is not really a fee.

Reply:
I did not change this now, but yes I find that too. That time I chose "fee" because `reward_per_storage_byte` etc summing up to `reward_per_execution` is confusing beccause it's different from the reward offered. Maybe `price_...`? I checked cloud providers pricing and they use fee